octodns/provider/ns1.py

```diff
@@ -216,6 +233,11 @@ class Ns1Provider(BaseProvider):
         # Only required if using dynamic records
         monitor_regions:
           - lga
+        # Optional. Default: false. true is Recommended, but not the default
+        # for backwards compatibility reasons. If true, all NS1 monitors will
+        # use a shared notify list rather than one per record & value
+        # combination.
+        shared_notifylist: false
```

Comment:
Does this need a more obvious warning about forward incompatibility? Not only in the comments/documentation but also the output of sync command?

If I create a record using the new version with shared_notifylist: true then downgrade and delete a pool/value, the old version will blindly knock out the associated notify list, breaking the other pools/records that were sharing it.

Reply:
> Does this need a more obvious warning about forward incompatibility? Not only in the comments/documentation but also the output of sync command?

Yes. I'd decided that `shared_notifylist: false` should at least be a warning saying that it's not the "preferred" way going forward and had been thinking over whether it should be a new required parameter to force people to decide on version update.  I think i've landed on the required parameter being too strong, but I'm open to others thoughts there.

> If I create a record using the new version with shared_notifylist: true then downgrade and delete a pool/value, the old version will blindly knock out the associated notify list, breaking the other pools/records that were sharing it.

Good call. That's a case I'd overlooked. Not coming up with great options thinking about it. Best I can come up with would be to release a version that leaves the shared ones alone on delete, but doesn't create them, but that'd only really allow updates to/from that version and you wouldn't be able to go back further so I'm not sure that buys anything.

Another option would be to declaire that once you enable `shared_notifylist` you can't go back versions.

Either way it might make sense to go ahead and add `extra_changes` bits to detect the state of notifylists and share/unshare them when they don't match the setting. That should give a controlled path to downgrade, though it could still allow someone to paint themselves into a corner if they were determined enough.

Reply:
I couldn't think of any good options either. At the very least, I think we should stick a log.warn if shared_notifylist is set to true and may be hyper-link (to this PR?) for a more detailed explanation.

> add extra_changes bits to detect the state of notifylists and share/unshare them

This will want to update every single dynamic record on running the sync command after setting the config to true. But I still like the idea. If someone wanted to knowingly downgrade, they have some way. With current implementation, there's no looking back.

Reply:
> This will want to update every single dynamic record on running the sync command after setting the config to true. But I still like the idea. If someone wanted to knowingly downgrade, they have some way. With current implementation, there's no looking back.

Yeah. In theory it'd be possible to make it smart enough to just update the monitors, but I haven't dug into how involved/possible that'd be.

Reply:
Added a CHANGELOG section with the cautions and caveots and linked that to this PR. Added a mention of the CHANGELOG in the parameter doc. Lmk what you think.